beacon-chain/blockchain/process_block.go

```diff
@@ -107,6 +107,11 @@ func (s *Service) onBlock(ctx context.Context, signed interfaces.ReadOnlySignedB
 		return err
 	}
 
+	// Verify that the parent block is in forkchoice
+	if !s.ForkChoicer().HasNode(b.ParentRoot()) {
+		return ErrNotDescendantOfFinalized
```

Comment:
A bit torn about this error message. It makes sense but there's some implementation details that may not be obvious for someone that's not in depth

Reply:
I am not sure how to change this, this error is exported and used in several places. Changing it on one side and not others seems odd, I can change it to "parent not in forkchoice" but that's even less info for someone not in depth, the only thing that can trigger this error is if the block does not connect to the finalized checkpoint

Reply:
Agree. I dont think it's a big deal but just wanted to point it out